sdk/provisioning/Azure.Provisioning/src/ModuleInfrastructure.cs

```diff
@@ -190,16 +194,21 @@ private void WriteConstructsByLevel(Queue<ModuleConstruct> constructs, string ou
             }
         }
 
-        private string GetFilePath(ModuleConstruct construct, string outputPath)
+        private string GetFilePath(ModuleConstruct? construct, string outputPath)
         {
-            string fileName = construct.IsRoot ? Path.Combine(outputPath, "main.bicep") : Path.Combine(outputPath, "resources", construct.Name, $"{construct.Name}.bicep");
+            string fileName = construct == null || construct.IsRoot ? Path.Combine(outputPath, "main.bicep") : Path.Combine(outputPath, "resources", construct.Name, $"{construct.Name}.bicep");
             Directory.CreateDirectory(Path.GetDirectoryName(fileName)!);
             return fileName;
         }
 
-        private void WriteBicepFile(ModuleConstruct construct, string outputPath)
+        private void WriteBicepFile(ModuleConstruct? construct, string outputPath)
         {
             using var stream = new FileStream(GetFilePath(construct, outputPath), FileMode.Create);
+            // just create an empty file if there is no construct
+            if (construct == null)
```

Comment:
Is passing null a valid scenario?  Why not just throw?

Reply:
In my mind, it makes more sense to not throw here because it isn't necessarily an error to not create resources, but don't have a strong preference either way.